libs/ledger/src/storage_unit/storage_unit_client.cpp

```diff
@@ -556,7 +539,7 @@ bool StorageUnitClient::Lock(ShardIndex index)
                                                     RevertibleDocumentStoreProtocol::LOCK);
 
     // wait for the promise
-    success = promise->As<bool>();
+    success = promise->GetResult(success) && success;
```

Comment:
This looks a bit suspect - if success were false and GetResult set it to true, would it still work as intended? In any case switching to success && promise->... would possibly be a minor time saver due to short circuit logic

Reply:
The key thing is that `GetResult()` populated `success` so it needs to be this way around. I agree a little confusing, but accurate